micro-rdk-nmea-macros/src/lib.rs

```diff
@@ -0,0 +1,60 @@
+pub(crate) mod attributes;
+pub(crate) mod composition;
+pub(crate) mod utils;
+
+use crate::composition::PgnComposition;
+use proc_macro::TokenStream;
+use proc_macro2::Span;
+use syn::DeriveInput;
+
+fn get_statements(input: &DeriveInput) -> Result<PgnComposition, TokenStream> {
```

Comment:
I was wondering why you have function that do not take self. It is impossible to have a Struct that holds all statements, fields etc... that can mutate self?